### PR TITLE
bump(*) - Use buildkit for dockerfile parsing

### PR DESCRIPTION
**Pull request by adambkaplan**
1. Move parser tests to use `moby/buildkit`'s parser
2. Use upstream version of `buildkit` instead of an outdated fork.
3. `go mod tidy && go mod vendor`

### CODE REVIEW

File: pkg/scripts/environment_test.go [hunk 1]
```diff
@@ -43,30 +41,30 @@ func equalArrayContents(a []string, b []string) bool {
 	return true
 }
 
-func TestConvertEnvironmentToDocker(t *testing.T) {
```
Comment (adambkaplan) [inline pkg/scripts/environment_test.go]: This test seems to run a pseudo-Docker build. It pulls in a lot of deps from `docker/docker` which seem to break things.

Moving this out for the time being.

File: pkg/build/strategies/onbuild/onbuild_test.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/docker/docker/builder/dockerfile/parser"
+	"github.com/moby/buildkit/frontend/dockerfile/parser"
```
Comment (adambkaplan) [inline pkg/build/strategies/onbuild/onbuild_test.go]: Use dockerfile parser from moby/buildkit

File: test/integration/dockerfile/dockerfile_test.go [hunk 3]
```diff
@@ -11,7 +11,8 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/docker/docker/builder/dockerfile/parser"
+	"github.com/moby/buildkit/frontend/dockerfile/parser"
```
Comment (adambkaplan) [inline test/integration/dockerfile/dockerfile_test.go]: Use dockerfile parser from moby/buildkit

File: pkg/scripts/environment_test.go [hunk 4]
```diff
@@ -43,30 +41,30 @@ func equalArrayContents(a []string, b []string) bool {
...
+// 		"`Value1`",
+// 		`"`,
+// 	}
+// 	var charValues []string
+// 	for ch := rune(32); ch < 127; ch++ {
+// 		charValues = append(charValues, "AB"+string(ch)+"CD", "AB\\"+string(ch)+"CD")
+// 	}
+// 	for _, testValue := range append(testValues, charValues...) {
+// 		list := api.EnvironmentList{{Name: "TEST", Value: testValue}}
+// 		converted := ConvertEnvironmentToDocker(list)
```
Comment (gabemontero) [inline pkg/scripts/environment_test.go]: Seems like you could do some validation of the return from `ConvertEnvironmentToDocker` without calling dockerfile functions

Comment (adambkaplan) [reply pkg/scripts/environment_test.go]: Yeah - the actual function is pretty trivial. I'll add a true unit test.

Comment (adambkaplan) [reply pkg/scripts/environment_test.go]: @gabemontero restored coverage via a simple unit test. PTAL

Comment (gabemontero) [reply pkg/scripts/environment_test.go]: cool

**Pull request merged**